esi/src/parse.rs

```diff
@@ -1,85 +1,118 @@
 use crate::{ExecutionError, Result};
 use log::debug;
-use quick_xml::events::BytesStart;
+use quick_xml::events::{BytesStart, Event as XmlEvent};
+use quick_xml::name::QName;
 use quick_xml::Reader;
 use std::io::BufRead;
+use std::ops::Deref;
+
+// State carrier of Try branch
+#[derive(Debug, PartialEq)]
+enum TryTagArms {
+    Try,
+    Attempt,
+    Except,
+}
 
 /// Representation of an ESI tag from a source response.
 #[derive(Debug)]
-pub enum Tag {
+pub struct Include {
+    pub src: String,
+    pub alt: Option<String>,
+    pub continue_on_error: bool,
+}
+
+#[derive(Debug)]
+pub enum Tag<'a> {
     Include {
         src: String,
         alt: Option<String>,
         continue_on_error: bool,
     },
+    Try {
+        attempt_events: Vec<Event<'a>>,
+        except_events: Vec<Event<'a>>,
+    },
 }
 
 /// Representation of either XML data or a parsed ESI tag.
 #[derive(Debug)]
 #[allow(clippy::upper_case_acronyms)]
 pub enum Event<'e> {
-    XML(quick_xml::events::Event<'e>),
-    ESI(Tag),
+    XML(XmlEvent<'e>),
+    ESI(Tag<'e>),
 }
 
-/// Parses the ESI document from the given `reader` and calls the `callback` closure upon each successfully parsed ESI tag.
-pub fn parse_tags<'a, R>(
-    namespace: &str,
+// #[derive(Debug)]
+struct EsiTags {
+    include: Vec<u8>,
+    comment: Vec<u8>,
+    remove: Vec<u8>,
+    tryy: Vec<u8>,
+    attempt: Vec<u8>,
+    except: Vec<u8>,
```

Comment:
The `tryy` is a smart workaround but I'm not a huge fan of the variable names not matching the known name of the tag. Perhaps we could prefix the struct keys with `tag_` instead?


```suggestion
    tag_include: Vec<u8>,
    tag_comment: Vec<u8>,
    tag_remove: Vec<u8>,
    tag_try: Vec<u8>,
    tag_attempt: Vec<u8>,
    tag_except: Vec<u8>,
```

Reply:
I had it this way, but then it should be referenced as `tag.tag_try` or  that makes this redundant. I think clippy is not happy about this too. :( since you insist... maybe I'll rename this to like `try_tag`, I think it's slightly better 😆 
I will have to rename it (everywhere), which will change this in a bunch of places. will push another commit.